esp-radio/src/wifi/mod.rs

```diff
@@ -2899,19 +2900,38 @@ impl Config {
     }
 }
 
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+/// Comprehensive error enum for Wi-Fi initialization failures.
+pub enum WifiInitError {
```

Comment:
maybe `InitializationError` should get folded into `WifiError` - since for BLE the Error type is something we really need to redesign it's maybe fine there